lib/MusicBrainz/Server/WebService/JSONSerializer.pm

```diff
@@ -14,53 +15,33 @@ sub fmt { 'json' }
 
 sub serialize
 {
-    my ($self, $type, @data) = @_;
+    my ($self, $type, $entity, $inc, $opts) = @_;
 
     $type =~ tr/-/_/;
 
-    my $override = $self->meta->find_method_by_name($type);
-    return $override->execute($self, @data) if $override;
+    if ($type =~ m/^(.*)_list$/ && exists($entity->{items})) {
+      if (exists($ENTITIES{$1}) && exists($ENTITIES{$1}->{plural_url})) {
+        $type = $1;
```

Comment:
Style issues:
- MBS uses four spaces (and never tabs) for each indentation level; at least in Perl code.
- The matched entity type deserves a proper (variable) name, IMO. I suggest writing `my ($entity_type) = $type =~ /.../`, then using `$entity_type` in this block.
- Not important: We mostly use plain `/.../` instead of `m/.../` in the existing codebase (by a 114:9 majority).